### PR TITLE
[F#] Extract Concepts from v2 exercise: bob

### LINKED ISSUE #262: [F#] Extract Concepts from v2 exercise: bob
Extract the list of concepts required to solve the v2 F# exercise: `bob`. For each concept, describe why it is required to solve the exercise. Keep in mind that an exercise can have multiple approaches. The goal of this task is to help determine the total list of concepts that make up `bob`.

The list of concepts should be stored in the following document: `languages/fsharp/reference/exercise-concepts/bob.md`. [Use this link to create a PR to add this document][pr-link].

[pr-link]: ../new/master?filename=languages/fsharp/reference/exercise-concepts/bob.md

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes #262

### CODE REVIEW

File: languages/fsharp/reference/exercise-concepts/bob.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Concepts of bob
+
+## General
+
+- functions: used as the main entry point for the exercise
+- function arguments: input strand is passed as an arguments
```
Comment (robkeim) [inline languages/fsharp/reference/exercise-concepts/bob.md]:
```suggestion
- function arguments: input strand is passed as an argument
```

File: languages/fsharp/reference/exercise-concepts/bob.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- methods: calling the `Char.IsLetter`, `String.IsNullOrEmpty` method and several `string` methods
```
Comment (robkeim) [inline languages/fsharp/reference/exercise-concepts/bob.md]:
```suggestion
- methods: calling the `Char.IsLetter`, `String.IsNullOrEmpty` method and several other `string` methods
```

File: languages/fsharp/reference/exercise-concepts/bob.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Concepts of bob
+
```
Comment (robkeim) [inline languages/fsharp/reference/exercise-concepts/bob.md]: Could you add a link here to the v2 implementation?

File: languages/fsharp/reference/exercise-concepts/bob.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+## Approach: pattern matching
+
+- pattern matching: matching on the input
+- guard clauses: adding guard clauses to the pattern matching
+- discards: ignore the pattern matches for the guard clauses to work
+
+## Approach: regular expressions
+
+- conditionals using if-elif-else: conditionally execute logic using an `if/elif/else` statement
+- regular expressions: checking the string types using regular expressions
```
Comment (robkeim) [inline languages/fsharp/reference/exercise-concepts/bob.md]: I think by "string types" here you mean question, yelling, etc? In that case, I feel "phrase types"  would be more explicit:
```suggestion
- regular expressions: checking the phrase types using regular expressions
```

File: languages/fsharp/reference/exercise-concepts/bob.md (hunk 3 above)
Comment (ErikSchierboom) [reply languages/fsharp/reference/exercise-concepts/bob.md]: Done!

**Pull request merged**